Sources/Spezi/Notifications/Spezi+RegisterRemoteNotificationsAction.swift

```diff
@@ -0,0 +1,106 @@
+            }
+
+            return try await spezi.remoteNotificationRegistrationSupport()
+        }
+    }
+
+    /// Registers to receive remote notifications through Apple Push Notification service.
+    ///
+    /// For more information refer to the [`registerForRemoteNotifications()`](https://developer.apple.com/documentation/uikit/uiapplication/1623078-registerforremotenotifications)
+    /// documentation for `UIApplication` or for the respective equivalent for your current platform.
+    ///
+    /// - Note: For more information on the general topic on how to register your app with APNs,
+    ///     refer to the [Registering your app with APNs](https://developer.apple.com/documentation/usernotifications/registering-your-app-with-apns)
+    ///     article.
+    ///
+    /// Below is a short code example on how to use this action within your ``Module``.
+    ///
+    /// - Warning: Registering for Remote Notifications on Simulator devices might not be possible if your are not signed into an Apple ID on the host machine.
+    ///     The method might throw a [`TimeoutError`](https://swiftpackageindex.com/stanfordspezi/spezifoundation/documentation/spezifoundation/timeouterror)
+    ///     in such a case.
+    ///
+    /// ```swift
+    /// import SpeziFoundation
+    ///
+    /// class ExampleModule: Module {
+    ///     @Application(\.registerRemoteNotifications)
+    ///     var registerRemoteNotifications
+    ///
+    ///     func handleNotificationsPermissions() async throws {
+    ///         // Make sure to request notifications permissions before registering for remote notifications
+    ///         try await UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound])
+    ///
+    ///
+    ///         do {
+    ///             let deviceToken = try await registerRemoteNotifications()
+    ///         } catch let error as TimeoutError {
+    /// #if targetEnvironment(simulator)
+    ///             return // override logic when running within a simulator
+    /// #else
+    ///             throw error
+    /// #endif
+    ///         }
+    ///
+    ///         // .. send the device token to your remote server that generates push notifications
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// > Tip: Make sure to request authorization by calling [`requestAuthorization(options:completionHandler:)`](https://developer.apple.com/documentation/usernotifications/unusernotificationcenter/requestauthorization(options:completionhandler:))
+    ///     to have your remote notifications be able to display alerts, badges or use sound. Otherwise, all remote notifications will be delivered silently.
+    ///
+    /// ## Topics
+    /// ### Action
+    /// - ``RegisterRemoteNotificationsAction``
+    @_disfavoredOverload
+    @available(*, deprecated, message: "Please migrate to the new SpeziNotifications package.")
```

Comment:
We should link to https://github.com/StanfordSpezi/SpeziNotifications here. Would be good to move the relevant code to the repo before we tag a next release with this depreciation warning.

Reply:
I added the link to the deprecation message 👍